src/resolve/resolved-dns-packet.c

```diff
@@ -2570,6 +2570,70 @@ bool dns_packet_equal(const DnsPacket *a, const DnsPacket *b) {
         return dns_packet_compare_func(a, b) == 0;
 }
 
+int dns_packet_ede_rcode(DnsPacket *p, uint16_t *ret_ede_rcode) {
+        assert(p);
+
+        bool has_ede = false;
+        uint16_t ede_rcode = _DNS_EDNS_OPT_INVALID;
```

Comment:
_DNS_EDNS_OPT_INVALID is defined as -EINVAL, i.e. -22, which is outside the uint16_t range. I'd just use `int` as type here to make things simple